src/solver/mha/mha_solver_backward.cpp

```diff
@@ -49,23 +47,26 @@ namespace mha {
 
 namespace { // TODO: Issue #2748
 
-MultiBufferWorkspaceTraits SplitBufferToWorkspace(size_t S, size_t D, size_t NHS)
+MultiBufferWorkspaceTraits
+SplitBufferToWorkspace(size_t S, size_t D, size_t NHS, miopenDataType_t out_type)
 {
     // the first MatMuls (N*H*S*D) * (N*H*S*D)T = (N*H*S*S)
     // the second MatMuls (N*H*S*S)[T] * (N*H*S*D) = (N*H*S*D)
     // dOxO row reduction (N*H*S*1)
-    return MultiBufferWorkspaceTraits{NHS * S * get_data_size(miopenFloat), // first matmuls
-                                      NHS * S * get_data_size(miopenFloat), // first matmuls
-                                      NHS * std::max(S, D) *
-                                          get_data_size(miopenFloat), // reduction and second matmul
-                                      NHS * D * get_data_size(miopenFloat),  // second matmuls
-                                      NHS * D * get_data_size(miopenFloat)}; // second matmuls
+    return MultiBufferWorkspaceTraits{
+        NHS * S * get_data_size(miopenFloat),                             // fp32 QxK and fp32 S
```

Comment:
off topic: I realized that `get_data_size()` is a duplicate of `GetTypeSize()` in tensor.hpp. We should delete the former in favor of the latter.

Reply:
Let me think about it. The problem is that I either have to pass tensor descriptor to that function and re-calculate all the parameters once again, or I can pass tensor descriptor just to get the TypeSize(), but actually I've already got the type earlier. It also breaks some sort of codestyle - some parts will be `sizeof(float)`, some others `GetTypeSize()` or `outputTypeSize`.
I'm not sure that there is just a simple replacement.